source/community.applications/usr/local/emhttp/plugins/community.applications/include/helpers.php

```
@@ -453,8 +468,20 @@ function caIsDockerRunning() {
 function writeJsonFile($filename,$jsonArray) {
 	debug(($_POST['action']??'Unknown')." - Write JSON File $filename");
-	$result = ca_file_put_contents($filename,json_encode($jsonArray,JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT));
+	$result = ca_file_put_contents($filename,serialize($jsonArray));
 	debug("Memory Usage:".round(memory_get_usage()/1048576,2)." MB");
+
+	// The plugin script needs a templates.json in JSON format to update support URLs on plugins
+	// If we're writing $templates, then save templates.json but filtered only for plugins to save space
+	if ( $filename == CA_PATHS['community-templates-info'] ) {
+		// array_values re-indexes to a flat JSON array - post_plugin_checks reads
+		// this with an index loop ($db[$i]), so the keys must be sequential.
+		ca_file_put_contents(CA_PATHS['community-templates-info-old'],json_encode(array_values(array_map(function($t) {
```

PHPStan (2.2.1)

[error] Constant CA_PATHS not found.
Learn more at https://phpstan.org/user-guide/discovering-symbols

(constant.notFound)